test/hotspot/jtreg/runtime/jni/daemonDestroy/TestDaemonDestroy.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+/*
+ * @test
+ * @bug 8290732
+ * @comment Test uses custom launcher that attempts to destroy the VM on both
+ *          a daemon and non-daemon thread. The result should be the same in
+ *          both cases.
+ * @requires vm.flagless
+ * @library /test/lib
+ * @build Main
+ * @run main/native TestDaemonDestroy
+ * @run main/native TestDaemonDestroy daemon
+ */
```

Comment:
If you do two @test sections, you can give them speaking names(e.g. @test id=demon) and they can run in parallel.

Reply:
I didn't think that was warranted/necessary here.

Reply:
I disagree. Splitting the test leads causes better runtime (on my 24 core machine: 10.5 seconds vs 16 seconds). It also makes the jtr file easier to read, since it only contains one test. It also is simpler to run a selected test if you want to reproduce an error.

It is regrettable that splitting the test section causes too much duplication. Maybe we could improve the jtreg syntax in the future.

Reply:
But you also have to duplicate the build instructions and building takes longer than execution on my systems (7 seconds versus 5) so if I had to pay for machine time I'd actually be worse off now!

Reply:
Oh, you are right. That's unfortunate.